pkg/controller/master-controller-manager/kubeone/controller.go

```diff
@@ -122,19 +200,121 @@ func (r *reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 		return reconcile.Result{}, err
 	}
 
-	err := r.reconcile(ctx, externalCluster)
+	if externalCluster.DeletionTimestamp != nil {
+		log.Debug("Deleting KubeOne Namespace")
+		ns := &corev1.Namespace{}
+		name := types.NamespacedName{Name: kubernetesprovider.GetKubeOneNameSpaceName(externalCluster.Name)}
+		if err := r.Get(ctx, name, ns); err != nil {
+			return reconcile.Result{}, err
+		}
+		if err := r.Delete(ctx, ns); err != nil {
+			return reconcile.Result{}, err
+		}
+
+		if err := kuberneteshelper.TryRemoveFinalizer(ctx, r, externalCluster, apiv1.ExternalClusterKubeOneNamespaceCleanupFinalizer); err != nil {
+			log.Errorf("Could not delete kubeone namespace, %w", err)
```

Comment:
This is misleading, this error is about not being able to remove a finalizer.

Also, please use structured logging: `log.Errorw("message", zap.Error(err))`